jsx/Form.js

```diff
@@ -258,9 +241,11 @@ class SearchableDropdown extends Component {
   componentDidUpdate(prevProps) {
     // need to clear out currentInput for when props.value gets wiped
     // if the previous value prop contained data and the current one doesn't
-    // clear currentInput
+    // clear currentInput.
     if (prevProps.value && !this.props.value) {
       this.setState({currentInput: ''});
+    } else if (this.props.value !== prevProps.value && this.props.value) {
+      this.setState({currentInput: this.props.options[this.props.value]});
```

Comment:
not in **24.1**

_This seems to be addition of code, rather than a replacement, which may be useful for a particular use-case._

Include as override and likely propose to **main**